logging/unit_tests/test_client.py

```diff
@@ -571,6 +571,70 @@ def test_list_metrics_with_paging(self):
             },
         })
 
+    def test_get_default_handler_app_engine(self):
+        import os
+        from google.cloud._testing import _Monkey
+        from google.cloud.logging.client import _APPENGINE_FLEXIBLE_ENV_VM
+        from google.cloud.logging.handlers import app_engine as _MUT
+        from google.cloud.logging.handlers import AppEngineHandler
+
+        client = self._makeOne(project=self.PROJECT,
+                               credentials=_Credentials(),
+                               use_gax=False)
+
+        os.environ[_APPENGINE_FLEXIBLE_ENV_VM] = "True"
```

Comment:
Don't modify `os.environ` in place (e.g. what if your test fails before reaching the cleanup). Use `_Monkey(os, environ={...})` instead